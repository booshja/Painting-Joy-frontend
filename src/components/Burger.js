// dependencies
import React, { useContext } from "react";
import styled from "styled-components";
import { breakpoints } from "../breakpoints";
// context
import MenuContext from "../context/MenuContext";

const StyledBurger = styled.button`
    /* position: absolute;
    top: 1rem;
    right: 1rem; */
    display: flex;
    flex-direction: column;
    justify-content: space-around;
    width: 2rem;
    height: 2rem;
    background: transparent;
    border: none;
    cursor: pointer;
    padding: 0;
    z-index: 101;

    &:focus {
        outline: none;
    }

    div {
        width: 2rem;
        height: 0.3rem;
        background-color: ${({ open }) => (open ? "#ffffff" : "#207070")};
        border-radius: 10px;
        transition: all 0.3s linear;
        position: relative;
        transform-origin: 1px;

        :first-child {
            transform: ${({ open }) => (open ? "rotate(45deg)" : "rotate(0)")};
        }

        :nth-child(2) {
            opacity: ${({ open }) => (open ? "0" : "1")};
            transform: ${({ open }) =>
                open ? "translateX(-20px)" : "translateX(0)"};
        }

        :nth-child(3) {
            transform: ${({ open }) => (open ? "rotate(-45deg)" : "rotate(0)")};
        }
    }

    /* @media (min-width: ${breakpoints.desktopSm}px) {
        display: none;
    } */
`;

const Burger = () => {
    // set up context
    const { menuOpen, setMenuOpen } = useContext(MenuContext);
    return (
        <StyledBurger open={menuOpen} onClick={() => setMenuOpen(!menuOpen)}>
            <div />
            <div />
            <div />
        </StyledBurger>
    );
};

export default Burger;
