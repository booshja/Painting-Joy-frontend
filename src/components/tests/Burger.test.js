import { render } from "@testing-library/react";
import Burger from "../Burger";
import MenuContext from "../../context/MenuContext";

/** Smoke Test */
it("renders without crashing, menuOpen true", () => {
    render(
        <MenuContext.Provider value={{ menuOpen: true, setMenuOpen: null }}>
            <Burger />
        </MenuContext.Provider>
    );
});

it("renders without crashing, menuOpen false", () => {
    render(
        <MenuContext.Provider value={{ menuOpen: false, setMenuOpen: null }}>
            <Burger />
        </MenuContext.Provider>
    );
});

/** Snapshot Test */
it("matches snapshot, menuOpen true", () => {
    const { asFragment } = render(
        <MenuContext.Provider value={{ menuOpen: true, setMenuOpen: null }}>
            <Burger />
        </MenuContext.Provider>
    );
    expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot, menuOpen false", () => {
    const { asFragment } = render(
        <MenuContext.Provider value={{ menuOpen: false, setMenuOpen: null }}>
            <Burger />
        </MenuContext.Provider>
    );
    expect(asFragment()).toMatchSnapshot();
});

/** RTL Tests */
it("renders correctly, menuOpen true", () => {
    const { getByTestId } = render(
        <MenuContext.Provider value={{ menuOpen: true, setMenuOpen: null }}>
            <Burger />
        </MenuContext.Provider>
    );

    getByTestId("div1");
    getByTestId("div2");
    getByTestId("div3");
});

it("renders correctly, menuOpen false", () => {
    const { getByTestId } = render(
        <MenuContext.Provider value={{ menuOpen: false, setMenuOpen: null }}>
            <Burger />
        </MenuContext.Provider>
    );

    getByTestId("div1");
    getByTestId("div2");
    getByTestId("div3");
});
