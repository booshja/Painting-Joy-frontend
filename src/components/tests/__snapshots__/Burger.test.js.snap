// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot, menuOpen false 1`] = `
<DocumentFragment>
  <button
    class="sc-bdnxRM cgpxYD"
  >
    <div
      data-testid="div1"
    />
    <div
      data-testid="div2"
    />
    <div
      data-testid="div3"
    />
  </button>
</DocumentFragment>
`;

exports[`matches snapshot, menuOpen true 1`] = `
<DocumentFragment>
  <button
    class="sc-bdnxRM hdArge"
    open=""
  >
    <div
      data-testid="div1"
    />
    <div
      data-testid="div2"
    />
    <div
      data-testid="div3"
    />
  </button>
</DocumentFragment>
`;
